src/core/imap.ts

```diff
@@ -165,28 +143,30 @@ export class IMAP {
 
                     await this.openBox(mailBox);
 
-                    const messages = await this.search(criteria);
-                    const newMessages : number[] = messages.filter((m) => ! this.uids.includes(m));
-
+                    const newMessages = await this.search(criteria);
                     console.log(newMessages, 'new Messages');
 
                     if (newMessages.length && !this.pending) {
                         this.pending = true;
-                        const data = await this.fetchAll(newMessages, {bodies: ''});
 
+                        setupTempFolder = new SetupTempFolder();
+                        setupTempFolder.createTempFolder();
+
+                        const data = await this.fetchAll(newMessages, {bodies: ''});
+                        await this.markAsSeen(newMessages);
                         for (let i = 0; i < data.length; i++) {
                             // post email
 
                         }
-                        // to save the id of converted messages.
-                        await this.updateConvertedMessages(newMessages);
-                        this.uids.push(...newMessages);
-
+                        setupTempFolder.removeTempFolder();
```

Comment:
Not sure if it will be any faster but you don't need to delete the whole folder, only specific files corresponding to already processed emails.

Reply:
Deleting an entire file is a little faster than looping through each file and deleting it.

>  only specific files corresponding to already processed emails.


This is actually only what will be in the file, which means that the whole file will be deleted, and I also want to leave no trace after doing what is required.

I tested it by converting more than 100 eml files. Some of them contain large files containing videos and images, and the conversion process works fine.